### DISCUSSION TITLE
Welcome to uapi-json Discussions!

### DISCUSSION THREAD

**Discussion by Smotrov** [Travelport-Ukraine/uapi-json#501] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by unnikrishnangs** [comment]
We have completely integrated the API using this package. Still we are being asked to work on the Transaction data for getting certification. Has anybody faced the same issue with this package? We really need some inputs on this issue.

**Comment by Smotrov** [comment]
Dear @unnikrishnangs! 
Thank you for using our library! 
From your message, I understand that you've been asked by Travelport to provide some information in order to get certified (which is actually a usual practice). 
Can I ask you to share with us which difficulties you faced? Which information are you missing?

**Comment by antu12** [comment]
HI, was trying out the terminal service and using sign on command I got an error. Is there anything I'm doing wrong or need to change?

Response SOAP:  <SOAP:Envelope 
  xmlns:SOAP="http://schemas.xmlsoap.org/soap/envelope/">
  <SOAP:Body>
    <terminal:TerminalRsp TransactionId="553B92CD0A0EE116095994A774B65908" ResponseTime="104" 
      xmlns:terminal="http://www.travelport.com/schema/terminal_v33_0">
      <terminal:TerminalCommandResponse>
        <terminal:Text>>PASSWORD? &lt;</terminal:Text>
      </terminal:TerminalCommandResponse>
    </terminal:TerminalRsp>
  </SOAP:Body>
</SOAP:Envelope>

TypeError: response.join is not a function
    at processResponse (......../node_modules/uapi-json/src/Services/Terminal/Terminal.js:165:16)
    at Object.executeCommand (....../node_modules/uapi-json/src/Services/Terminal/Terminal.js:248:32)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

**Reply by kommandant-topp** [reply]
Hello @antu12,
Could you add xml request of your attempt?

**Reply by dchertousov** [reply]
Hello @antu12 !

There is no need to use SIGN-ON command in uAPI terminal, as your access credentials already define your access rights and PCC you work in.

**Reply by antu12** [reply]
Thanks for the reply, 
Well I was using the TerminalService
so after running the example-> interactive.js  when I was running any command I was asked to Sign on. So if  I run the command with my sign on name "SON/***" the response is as below,
Response SOAP: <SOAP:Envelope
xmlns:SOAP="[http://schemas.xmlsoap.org/soap/envelope/">](http://schemas.xmlsoap.org/soap/envelope/%22%3E)
SOAP:Body
<terminal:TerminalRsp TransactionId="553B92CD0A0EE116095994A774B65908" ResponseTime="104"
xmlns:terminal="[http://www.travelport.com/schema/terminal_v33_0">](http://www.travelport.com/schema/terminal_v33_0%22%3E)
terminal:TerminalCommandResponse
terminal:Text>PASSWORD? <</terminal:Text>
</terminal:TerminalCommandResponse>
</terminal:TerminalRsp>
</SOAP:Body>
</SOAP:Envelope>

then I get the error,
TypeError: response.join is not a function
at processResponse (......../node_modules/uapi-json/src/Services/Terminal/Terminal.js:165:16)
at Object.executeCommand (....../node_modules/uapi-json/src/Services/Terminal/Terminal.js:248:32)
at processTicksAndRejections (internal/process/task_queues.js:93:5)

Here in Terminal.js:165:16 its response.join("\n") is failing as according to the response we are getting "PASSWORD? " string.
and not an array of string.

So I was just asking that do we not use the terminal for the sign-on command and if there was some other way to sign-on.

**Reply by antu12** [reply]
> Hello @antu12 !
> 
> There is no need to use SIGN-ON command in uAPI terminal, as your access credentials already define your access rights and PCC you work in.

Hello @dchertousov,

But every time I run the terminal service it gives me response to >SIGN IN <
So does it mean my credentials are not created properly?

**Reply by dchertousov** [reply]
@antu12 I can only advise to contact TVPT support, as we are providing the support for the library only and can not resolve the issues with credentials.

Please use https://travelportprod.service-now.com/mytravelport_support/ service with credentials, provided by trvaelport local authorities or the affiliates to receive help on this matter

**Reply by antu12** [reply]
@dchertousov Sure.
Thanks a lot. I didn't knew the info you just provided, that alone is a lot of help.

**Comment by timmyTurner013** [comment]
Hello
I was able to run your program as a terminal session. 
I had a query about this implementation. 
I have attached a screenshot below to explain my query. 
In Travelport Smartpoint, the data in the red box of the image is clickable. 
I wanted to know if there is any way to fetch data for these data respectively and make my terminal session clickable as well. 
![term1](https://user-images.githubusercontent.com/90834169/231722724-e3799799-ecb7-4d25-8d1d-f9bf748cdd76.JPG)

Thank you.
